internal/ethapi/transaction_args.go

```diff
@@ -266,15 +266,15 @@ func (args *TransactionArgs) ToMessage(globalGasCap uint64, baseFee *big.Int) (t
 
 // toTransaction converts the arguments to a transaction.
 // This assumes that setDefaults has been called.
-func (args *TransactionArgs) toTransaction() *types.Transaction {
+func (args *TransactionArgs) toTransaction(isGingerbreadP2 bool) *types.Transaction {
 	var data types.TxData
 	switch {
 	case args.MaxFeePerGas != nil:
 		al := types.AccessList{}
 		if args.AccessList != nil {
 			al = *args.AccessList
 		}
-		if args.FeeCurrency != nil || args.GatewayFeeRecipient != nil || args.GatewayFee != nil {
+		if args.GatewayFeeRecipient != nil || args.GatewayFee != nil || (!isGingerbreadP2 && args.FeeCurrency != nil) {
```

Comment:
Mh, doesn't this mean that no `CeloDynamicFeeTx` will ever be created when `isGingerbreadP2` is true?
This deprecation and not what we want, right?

Reply:
Ok, get it now. Makes sense.